### PR TITLE
Add settings dialog starter

### PR DESCRIPTION
**Pull request by bbogdanov**
Starting small with only one single setting available.

image

### CODE REVIEW

File: src/components/dialogs/settingsDialog/DownloadSettings.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/* eslint-disable vue/max-attributes-per-line */
```
Comment (CzBiX) [inline src/components/dialogs/settingsDialog/DownloadSettings.vue:1]: Can be removed?

File: src/components/dialogs/settingsDialog/DownloadSettings.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  },
+})
+export default class DownloadSettings extends Vue {
+  preferences!: Preferences;
+
+  updatePreferencesRequest!: (_: any) => void;
+
+  changeSettings(property: string, value: string | boolean) {
+    // copy of preferences with changed certrain property and its value
+    const updatedPreferences = Object.assign({}, this.preferences, {
```
Comment (CzBiX) [inline src/components/dialogs/settingsDialog/DownloadSettings.vue:49]: It's possible to just post changed preference.

File: src/components/dialogs/settingsDialog/SettingsDialog.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/* eslint-disable vue/max-attributes-per-line */
```
Comment (CzBiX) [inline src/components/dialogs/settingsDialog/SettingsDialog.vue:1]: same above

**Review by CzBiX — CHANGES_REQUESTED** [review]

File: src/components/dialogs/settingsDialog/DownloadSettings.vue (hunk 2 above)
Comment (bbogdanov) [reply src/components/dialogs/settingsDialog/DownloadSettings.vue:49]: I am not sure what are you suggesting. Can you please explain?

File: src/store/index.ts [hunk 4]
```diff
@@ -67,6 +68,9 @@ const store = new Vuex.Store<RootState>({
     /* eslint-enable no-param-reassign */
   },
   getters: {
+    allPreferences(state) {
```
[the remark this answers is not in the document]
Comment (bbogdanov) [reply src/store/index.ts:71]: There is no hidden intention. Just following what I did in the search plugin form

File: src/components/dialogs/settingsDialog/DownloadSettings.vue (hunk 2 above)
Comment (CzBiX) [reply src/components/dialogs/settingsDialog/DownloadSettings.vue:49]: Please see note 1 in https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-(qBittorrent-4.1)#set-application-preferences

Comment (bbogdanov) [reply src/components/dialogs/settingsDialog/DownloadSettings.vue:49]: Got it!

**Pull request merged**